assets/src/edit-story/components/library/panes/media/common/paginatedMediaGallery.js

```diff
@@ -77,30 +83,61 @@ function PaginatedMediaGallery({
       currentPaddingLeft - scrollbarWidth + 'px';
   }, [scrollbarWidth, refContainer]);
 
-  const refContainerFooter = useRef();
-  useIntersectionEffect(
-    refContainerFooter,
-    {
-      root: refContainer,
-      // This rootMargin is added so that we load an extra page when the
-      // "loading" footer is "close" to the bottom of the container, even if
-      // it's not yet visible.
-      rootMargin: `0px 0px ${ROOT_MARGIN}px 0px`,
-    },
-    (entry) => {
-      if (
-        !isMediaLoaded ||
-        isMediaLoading ||
-        !hasMore ||
-        !entry.isIntersecting
-      ) {
+  const [handleScroll] = useDebouncedCallback(
+    (e) => {
+      if (!hasMore || !isMediaLoaded || isMediaLoading) {
         return;
       }
-      setNextPage();
+      // This rootMargin is added so that we load an extra page when the
+      // we are "close" to the bottom of the container, even if it's not
+      // yet visible.
+      const bottom =
+        e.target.scrollHeight - e.target.scrollTop <=
+        e.target.clientHeight + ROOT_MARGIN;
+      if (bottom) {
+        setNextPage();
+      }
     },
-    [hasMore, isMediaLoading, isMediaLoaded, setNextPage]
+    500,
+    [hasMore, isMediaLoaded, isMediaLoading, setNextPage]
   );
 
+  useEffect(() => {
+    const node = refContainer.current;
+    if (!node) {
+      return undefined;
+    }
+
+    // When the user scrolls (debounced).
+    node.addEventListener('scroll', handleScroll);
+    return () => node.removeEventListener('scroll', handleScroll);
+  }, [handleScroll]);
+
+  useEffect(() => {
+    if (!hasMore || !isMediaLoaded || isMediaLoading) {
+      return;
+    }
+
+    //eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+    const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+    async function setNextPageIfCurrentPageNotFull() {
+      // Wait for <Gallery> to finish rendering first.
+      await sleep(50);
```

Comment:
Should we do another `isLoading` etc check after this sleep? (would it be possible for both `setNextPage()`s to be triggered?)

Reply:
Done.